client/signup/jetpack-connect/index.jsx

```diff
@@ -234,10 +229,16 @@ class JetpackConnectMain extends Component {
 		if ( ! this.checkProperty( 'isJetpackActive' ) ) {
 			return 'notActiveJetpack';
 		}
-		if ( ! this.checkProperty( 'isJetpackConnected' ) ) {
+		if (
+			! this.checkProperty( 'isJetpackConnected' ) ||
+			( this.checkProperty( 'isJetpackConnected' ) && ! this.checkProperty( 'userOwnsSite' ) )
```

Comment:
> When I connect a site, then try to connect again with a different user, I wind up with the notice for alreadyConnectedByOtherUser.

Can you list out the steps to wind up in that state? Did you fully log out, or use an incognito window when trying to link an other user?